src/server/controllers/index.js

```diff
@@ -58,7 +58,31 @@ function filterResponseFields(req, data, options) {
 }
 
 function filterResponseFieldsAll(req, arr, options) {
-  return arr.map(data => {
+  if (req.locals && req.locals.graphData) {
```

Comment:
Will we need the filtering of the graph output somewhere else as well / in another controller? If not we should maybe move in in an own helper method etc. as it might be too specific for a base controller.

Reply:
This is why I sent it to you ... I did it this way because I wanted to take advantage of the pagination/response formatting mostly

I do think we might want to combine data from the graph and our db in other places, but I don't think the top3 specifically is likely to be repeated. Right now the threshold-filtering and the graph-combining are separate, but the graph-combining needs to happen before the threshold filter. And I also want the threshold filter to interact with the protected fields, so that this endpoint returns everything for an authenticated user 

I guess I'm not sure where better to put the helper method

Reply:
Hmn ja, I see! I guess we need something more composable here (like a way to combine different sub methods, similar to a middleware), but that could also be an overkill in another way. Let's keep it here for now and just add a few comments so its easier to understand whats going on?